src/bin/nydus-image/deduplicate.rs

```diff
@@ -0,0 +1,239 @@
+                chunk_blob_id    TEXT NOT NULL,
+                chunk_digest     TEXT,
+                chunk_compressed_size  INT,
+                chunk_uncompressed_size  INT,
+                chunk_compressed_offset  INT,
+                chunk_uncompressed_offset  INT
+            )",
+            [],
+        )?;
+        Ok(())
+    }
+
+    fn insert_data_into_table(conn: &Connection, chunk_table: &ChunkTable) -> Result<()> {
+        conn.execute(
+            "INSERT INTO chunk(chunk_blob_id,chunk_digest,chunk_compressed_size,
+                chunk_uncompressed_size,chunk_compressed_offset,chunk_uncompressed_offset)
+            VALUES (?1, ?2, ?3, ?4, ?5, ?6);
+            ",
+            params![
+                chunk_table.chunk_blob_id,
+                chunk_table.chunk_digest,
+                chunk_table.chunk_compressed_size,
+                chunk_table.chunk_uncompressed_size,
+                chunk_table.chunk_compressed_offset,
+                chunk_table.chunk_uncompressed_offset
+            ],
+        )?;
+
+        Ok(())
+    }
+
+    fn get_data_from_table(conn: &Connection) -> Result<Vec<ChunkTable>> {
+        let mut stmt = conn.prepare(
+            "SELECT id, chunk_blob_id, chunk_digest,chunk_compressed_size,
+        chunk_uncompressed_size, chunk_compressed_offset, chunk_uncompressed_offset from chunk",
+        )?;
+        let chunk_iterator = stmt.query_map([], |row| {
+            Ok(ChunkTable {
+                id: row.get(0)?,
+                chunk_blob_id: row.get(1)?,
+                chunk_digest: row.get(2)?,
+                chunk_compressed_size: row.get(3)?,
+                chunk_uncompressed_size: row.get(4)?,
+                chunk_compressed_offset: row.get(5)?,
+                chunk_uncompressed_offset: row.get(6)?,
+            })
+        })?;
+        let mut chunks = Vec::new();
+        for chunk in chunk_iterator {
+            chunks.push(chunk?);
+        }
+        Ok(chunks)
+    }
+}
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub struct BlobTable {
+    // id: i32,
+    blob_id: String,
+    blob_size: String,
+    image_name: String,
+}
+
+impl Table for BlobTable {
+    fn clear(conn: Connection) -> Result<()> {
+        let _ = conn.execute("DROP TABLE blob", [])?;
+        Ok(())
+    }
+
+    fn creat_table(conn: &Connection) -> Result<()> {
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS blob (
+                blob_id             TEXT PRIMARY KEY ,
+                blob_size           TEXT NOT NULL,
+                image_name          TEXT
+            )",
+            [],
+        )?;
+        Ok(())
+    }
+
+    fn insert_data_into_table(conn: &Connection, blob_table: &BlobTable) -> Result<()> {
+        conn.execute(
+            "INSERT INTO blob (blob_id, blob_size, image_name)
+            SELECT ?1, ?2, ?3
+            WHERE NOT EXISTS (
+                SELECT blob_id
+                FROM blob
+                WHERE blob_id = ?4
```

Comment:
Is it possible to use `WHERE blob_id = ?1` here?

Reply:
Modified.